odoo/fields.py

```diff
@@ -461,18 +453,18 @@ def _get_attrs(self, model_class, name):
                 warnings.warn(f"precompute attribute has no impact on non stored field {self}")
                 attrs['precompute'] = False
         if attrs.get('company_dependent'):
-            # by default, company-dependent fields are not stored, not computed
-            # in superuser mode and not copied
-            attrs['store'] = False
-            attrs['compute_sudo'] = attrs.get('compute_sudo', False)
+            if attrs.get('required'):
+                warnings.warn(f"company_dependent field {self} cannot be required")
+            if attrs.get('translate'):
+                warnings.warn(f"company_dependent field {self} cannot be translated")
+            allowed_types = ('char', 'float', 'boolean', 'integer', 'text', 'many2one', 'date', 'datetime', 'selection', 'html')
```

Comment:
This should be moved as a constant at the beginning of the file.
This tuple is copied in the implementation of `ir.model.fields`, if I am not mistaken.  It should use a constant instead.